subsys/net/lib/lwm2m/lwm2m_obj_server.c

```diff
@@ -21,7 +21,13 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #endif
 
 #define SERVER_VERSION_MAJOR 1
+#if defined(CONFIG_LWM2M_RD_CLIENT_SUPPORT_BOOTSTRAP)
```

Comment:
I don't think this is right, the object version should not depend on whether bootstrap is enabled or not. It should rather be controlled by a separate Kconfig option, like IPSO objects do:
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/lib/lwm2m/Kconfig.ipso#L27